src/attr/setters.pyi

```diff
@@ -0,0 +1,12 @@
+from . import _OnSetAttrType, Attribute
+from typing import TypeVar, Any, NewType, cast
+
+_T = TypeVar("_T")
+
+def pipe(*setters: _OnSetAttrType) -> _OnSetAttrType: ...
+def frozen(instance: Any, attribute: Attribute, new_value: Any) -> Any: ...
+def validate(instance: Any, attribute: Attribute, new_value: _T) -> _T: ...
+def convert(instance: Any, attribute: Attribute, new_value: _T) -> _T: ...
+
+DisableType = NewType("DisableType", object)
```

Comment:
You might want to warn that this isn't a runtime type.  Maybe even make it `_DisableType` to enforce that.

Reply:
Yeah, thanks!